src/mlpack/core/data/string_cleaning.hpp

```diff
@@ -0,0 +1,126 @@
+  *
+  * @param input Vector of strings
+  */
+  void RemovePunctuation(std::vector<std::string>& input,
+                         const std::string& punctuation = "!\"#$%&'()*+"
+                         ",-./:;<=>?@[\\]^_`{|}~")
+  {
+    for (auto& str : input)
+    {
+      str.erase(boost::remove_if(str, boost::is_any_of(punctuation)),
+          str.end());
+    }
+  }
+
+  /**
+  * Function to remove a set of chars from a given vector of strings.
+  * A function which accepts a function as an argument and removes the
+  * character if the funtion returns true.
+  * 
+  * @param input Vector of strings.
+  * @param fun A function which takes const char& c as input and returns bool;
+  *
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be of bool type.
+  *
+  * Definiation of function should be of type
+  * bool fn(const char& c)
+  */
+  template<typename FunctionType>
+  void RemoveChar(std::vector<std::string>& input, const FunctionType& fun)
+  {
+    for (auto& str : input)
+      str.erase(boost::remove_if(str, fun), str.end());
+  }
+
+  /**
+  * Function to convert given vector of strings to lower case.
+  *
+  * @param input Vector of strings
+  */
+  void LowerCase(std::vector<std::string>& input)
+  {
+    for (auto& str : input)
+      boost::algorithm::to_lower(str);
+  }
+  /**
+  * Function to convert given vector of strings to Upper case.
+  *
+  * @param input Vector of strings
+  */
+  void UpperCase(std::vector<std::string>& input)
+  {
+    for (auto& str : input)
+      boost::algorithm::to_upper(str);
+  }
+  /**
+  * Function to remove stopwords from a given vector of strings.
+  *
+  * @param input Vector of strings
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  */
+  template<typename TokenizerType>
+  void RemoveStopWords(std::vector<std::string>& input,
+                       const std::unordered_set<boost::string_view,
+                                boost::hash<boost::string_view> >& stopwords,
+                       const TokenizerType& tokenizer)
+  {
+    for (auto& str : input)
+    {
+      std::string result = "";
+      boost::string_view token;
+      boost::string_view strView(str);
+      token = tokenizer(strView);
+      if (!token.empty() && stopwords.find(token) == stopwords.end())
+        result += std::string(token);
+      if (!token.empty())
+        token = tokenizer(strView);
+      std::cout<<"here"<<result<<"opt \n";
+      while (!token.empty())
+      {
+        if (stopwords.find(token) == stopwords.end())
+          result += (result == "") ? std::string(token) : " " + 
```

Comment:
```
result += (result == "") ? std::string(token) : " " + std::string(token);
```

The comparison is not needed. The first token has been written outside of the loop.
It's enough to write

```
result += " " + std::string(token);
```

Reply:
Take the case of `isn't Mlpack great`

If the first token is a Stop word then result will be empty and then it will have a space in front `result += " " + std::string(token)` , and ans will be ` Mlpack great`

Reply:
Yes, you're right.